src/test/xmeta/test_xmeta_emit.cpp

```diff
@@ -1115,24 +1292,310 @@ TEST_CASE("Interface type metadata")
+        { ExpectedEvent{ "Changed", "StringListEvent" } },
+        {
+            { ExpectedMethod("add_Changed", "Foundation.EventRegistrationToken", { ExpectedParam{"value", "StringListEvent"} }, interface_method_event_attributes()), "I1.add_Changed" },
+            { ExpectedMethod("remove_Changed", ElementType::Void, { ExpectedParam{"value", "Foundation.EventRegistrationToken"} }, interface_method_event_attributes()), "I1.remove_Changed" },
+            { ExpectedMethod("m0", ElementType::Void, {}, class_method_attributes()), "I1.m0" },
+            { ExpectedMethod("get_property1", ElementType::I4, {}, class_method_property_attributes()), "I1.get_property1" },
+            { ExpectedMethod("put_property1", ElementType::Void, { ExpectedParam{"value", ElementType::I4} }, class_method_property_attributes()), "I1.put_property1" }
+        }
+    };
+
+    ExpectedInterface i1{ "I1",
+            { ExpectedProperty("property1", ElementType::I4) },
+            { ExpectedEvent("Changed", "StringListEvent") },
+            {
+                ExpectedMethod("add_Changed", "Foundation.EventRegistrationToken", { ExpectedParam{"value", "StringListEvent"} }, interface_method_event_attributes()),
+                ExpectedMethod("remove_Changed", ElementType::Void, { ExpectedParam{"value", "Foundation.EventRegistrationToken"} }, interface_method_event_attributes()),
+                ExpectedMethod("m0", ElementType::Void, {}, interface_method_attributes()),
+                ExpectedMethod("get_property1", ElementType::I4, {}, interface_method_property_attributes()),
+                ExpectedMethod("put_property1", ElementType::Void, { ExpectedParam{"value", ElementType::I4} }, interface_method_property_attributes())
+            }
+    };
+
+    ExpectedDelegate d1{ "StringListEvent", ElementType::Void,
+       { ExpectedParam("sender", ElementType::I4) }
+    };
+
+    std::vector<std::shared_ptr<ExpectedType>> fileTypes =
+    {
+        std::make_unique<ExpectedClass>(c1),
+        std::make_unique<ExpectedInterface>(i1),
+        std::make_unique<ExpectedDelegate>(d1)
+    };
+    ValidateTypesInMetadata(test_idl, fileTypes);
+}
+
+TEST_CASE("Runtimeclass extend class base metadata")
+{
+    std::istringstream test_idl{ R"(
+        namespace N
+        {
+            runtimeclass C0
+            {
+            }
+
+            runtimeclass C1 : C0
+            {
+                Int32 Draw(Int32 param1);
+            }
+        }
+    )" };
+
+    ExpectedClass c0{ "C0", {}, {}, {}, {}};
+
+    ExpectedClass c1{ "C1", {}, {}, {}, {
+        { ExpectedMethod{ "Draw", ElementType::I4, { ExpectedParam{ "param1", ElementType::I4 } }, class_method_attributes() }, "IC1.Draw" },
+    }, "N.C0"};
+
+    ExpectedInterface ic1{ "IC1",
+        { ExpectedMethod{ "Draw", ElementType::I4, { ExpectedParam{ "param1", ElementType::I4 } }, interface_method_attributes() } }
+};
+
+    std::vector<std::shared_ptr<ExpectedType>> fileTypes =
```

Comment:
curious about the pattern - who are you sharing with?  why not unique_ptr?

Reply:
I was just thinking to turn all of them into non pointer types or just unique_ptrs like you said since it is just mostly going to be contained in a test. But I was racing against the clock.

Reply:
Okay I tried, ctrl+f replace std::shared_ptr<ExpectedType> to unique_ptr. Getting C2280 error. I am guessing somewhere deep in this code it is being shared somehow?